app/assets/javascripts/discourse/app/services/desktop-notifications.js

```diff
@@ -71,7 +70,7 @@ export default class DesktopNotificationsService extends Service {
 
   get isEnabledDesktop() {
     if (this.isGrantedPermission) {
-      return this.notificationsDisabled;
+      return !this.notificationsDisabled;
```

Comment:
This is where the bug was -- we check if `this.notificationsDisabled` is truthy... and `disabled` string is truthy..